src/Common/Logger.h

```diff
@@ -1,70 +1,160 @@
-  * If Logger already exists, throws exception.
-  * Logger is destroyed, when last shared ptr that refers to Logger with specified name is destroyed.
-  */
-LoggerPtr createLogger(const std::string & name, Poco::Channel * channel, Poco::Message::Priority level = Poco::Message::PRIO_INFORMATION);
+    DB::QuillLoggerPtr getQuillLogger();
 
-/** Create raw Poco::Logger that will not be destroyed before program termination.
-  * This can be used in cases when specific Logger instance can be singletone.
-  *
-  * For example you need to pass Logger into low-level libraries as raw pointer, and using
-  * RAII wrapper is inconvenient.
-  *
-  * Generally you should always use getLogger functions.
-  */
+    void setLogLevel(const std::string & level);
+    void setLogLevel(quill::LogLevel level);
+    void flushLogs();
 
-LoggerRawPtr getRawLogger(const std::string & name);
+    std::string_view getName()
+    {
+        return name;
+    }
 
-LoggerRawPtr createRawLogger(const std::string & name, Poco::Channel * channel, Poco::Message::Priority level = Poco::Message::PRIO_INFORMATION);
+    static DB::TextLogSink & getTextLogSink();
+    static OwnPatternFormatter * getFormatter();
+    static void setFormatter(std::unique_ptr<OwnPatternFormatter> formatter);
+
+    static void enableSyncLogging();
+    static bool shouldSyncLog();
+private:
+    std::string_view name;
+    std::string name_holder;
+    DB::QuillLoggerPtr logger;
+};
+
+using LoggerPtr = std::shared_ptr<Logger>;
+using LoggerRawPtr = Logger *;
+
+enum class LoggerComponent
+{
+    Root = 0,
+    RaftInstance,
+    ZooKeeperClient,
+    Max
+};
 
-/** Returns true, if currently Logger with specified name is created.
-  * Otherwise, returns false.
-  */
-bool hasLogger(const std::string & name);
+LoggerPtr getLogger(LoggerComponent component);
+LoggerPtr getLogger(const char * name, LoggerComponent component = LoggerComponent::Root);
+LoggerPtr getLogger(std::string_view name, LoggerComponent component = LoggerComponent::Root);
+LoggerPtr getLogger(std::string name, LoggerComponent component = LoggerComponent::Root);
+
+template <size_t n>
+LoggerPtr getLogger(const char (&name)[n])
+{
+    return getLogger(std::string_view{name, n});
+}
+
+DB::QuillLoggerPtr getQuillLogger(LoggerComponent component);
+DB::QuillLoggerPtr getQuillLogger(const std::string & name);
+
+LoggerPtr createLogger(const std::string & name, std::vector<std::shared_ptr<quill::Sink>> sinks);
+LoggerPtr createRootLogger(std::vector<std::shared_ptr<quill::Sink>> sinks);
+
+LoggerPtr getRootLogger();
 
 void disableLogging();
 
 bool isLoggingEnabled();
+
+
+/// This wrapper helps to avoid too frequent and noisy log messages.
+/// For each pair (logger_name, format_string) it remembers when such a message was logged the last time.
+/// The message will not be logged again if less than min_interval_s seconds passed since the previously logged message.
+class LogFrequencyLimiterImpl
```

Comment:
Maybe we can move this and `LogSeriesLimiter` to a different file to remove deps from Logger.h? I know it was in a bad place already.